### ISSUE TITLE
bora apply stack (update stack) error with 'on_failure' params

### ISSUE THREAD

**Issue by starchx** [ampedandwired/bora#31]
Create stack works fine but apply stack (update stack) failed with error: 
```
aws-sdk-core/param_validator.rb:28:in `validate!': unexpected value at params["on_failure"] (ArgumentError)
```
'on_failure' is not an argument for aws update-stack command but would be good for Bora to not pass this to update-stack.

ruby: 2.4.1p111 (windows)
gem: 2.6.11
aws-sdk-core-2.9.38

**Comment by herebebogans** [comment]
Hi @starchx 

That's a result of my recent PR #27.

I will have a look - do you have a sample bora.yml that you were using?

**Comment by ampedandwired** [comment]
This is happening because the filtering out of invalid update attributes in `cfn/stack.rb` assumes that keys are symbols, whereas the ones passed through from bora.yml are strings. Spec didn't pick it up because it uses symbols too.

Possible fixes:
1. Make `cfn/stack.rb` indifferent to strings and symbols (the aws-sdk accepts either, so this is a reasonable approach I think.
3. Make sure all keys passed in to `cfn/stack.rb` are symbolized.
3. Make sure all keys passed in to `cfn/stack.rb` are stringified.

Reviewing the code in `cfn/stack.rb`, there are a few other places this could potentially happen too.

**Comment by ampedandwired** [comment]
I think the other methods in `cfn/stack.rb` are OK.

Generally we use symbols throughout the code for the params passed into `cfn/stack.rb`, so possibly the best fix would be to convert any string keys to symbols in `cfn_options_from_stack_config`.

**Comment by ampedandwired** [comment]
Fixed by symbolizing the keys being passed into `cfn/stack.rb`. Released in 1.7.2.

**Comment by starchx** [comment]
@ampedandwired Awesome! Thanks for the quick fix. bora apply now works. 
bora diff still complains the same error though, but less important. It still works.

**Comment by ampedandwired** [comment]
Ah, sorry about that. Will reopen and fix the diff too.

**Comment by ampedandwired** [comment]
Fixed in 1.7.3. I'm going to change the way this works a bit and add some integration tests to catch stuff like this in the future.

**Comment by ampedandwired** [comment]
Fixed in 1.7.3. I'm going to change the way this works a bit and add some integration tests to catch stuff like this in the future.
